mk/config.mk

```diff
@@ -275,6 +275,16 @@ CFG_DT ?= n
 # editing of the supplied DTB.
 CFG_DTB_MAX_SIZE ?= 0x10000
 
+# Device Tree Overlay support.
+# This define enables support for an OPTEE provided a DTB overlay.
+# One of two modes is supported in this case:
+# 1. Append OPTEE nodes to an existing DTB overlay located at CFG_DT_ADDR or
+#    passed in arg2
+# 2. Generate a new DTB overlay at CFG_DT_ADDR
+# A subsequent boot stage must then merge the generated overlay DTB into a main
+# DTB using the standard fdt_overlay_apply() method.
```

Comment:
If FDT address boot argument  (`r2` or `CFG_DT_ADDR`) provides a full DTB, not an overlay DTB, should OP-TEE call `fdt_overlay_apply()` for its added fragments to be taken into account?

Reply:
I don't currently propose supporting that

1. Provide a full DTB and add nodes directly
2. Provide an overlay from OPTEE 
3. Append OPTEE overlay nodes to an existing overlay passed as a parameter

Reply:
ok. I wonder if OP-TEE should check whether the provided DTB blob is a DTB or a DTB overlay, or an uninited buffer, prior inserting the OP-TEE fragments. Do you expect `CFG_DT_OVERLAY=y` to implicitly expect DT overlays blobs  only from `r2` boot arg or `CFG_DT_ADDR`?

Reply:
Ok, I agree simpler is better. I suggest then to rename the config to something like `CFG_EXTERNAL_DTB_OVERLAY` or `CFG_EXTERNAL_DTB_OVERLAYS`, as opposed to `CFG_EMBEDDED_DT*` stuff.

Reply:
Fair enough CFG_EXTERNAL_DTB_OVERLAY